x/concentrated-liquidity/internal/math/tick.go

```diff
@@ -1,52 +1,165 @@
+	}
+
+	// The formula is as follows: k_increment_distance = 9 * 10**(-k_at_price_1)
+	// Due to sdk.Power restrictions, if the resulting power is negative, we take 9 * (1/10**k_at_price_1)
+	kIncrementDistance := sdkNineDec.Mul(handleNegativeExponents(kAtPriceOne.Neg()))
+
+	// If the price is less than 1, we must reduce the increment distance by a factor of 10
+	if price.LT(sdk.OneDec()) {
+		kIncrementDistance = kIncrementDistance.Quo(sdkTenDec)
+	}
+
+	// Initialize the total price to 1, the current k to k_at_price_1, and the number of ticks passed to 0
+	totalPrice := sdk.OneDec()
+	ticksPassed := sdk.ZeroInt()
+	currentK := kAtPriceOne
+
+	// Set the currentIncrement to the kAtPriceOne
+	curIncrement := handleNegativeExponents(kAtPriceOne)
+
+	// Now, we loop through the k increments until we have passed the price
+	// Once we pass the price, we can determine what which k values we have filled in their entirety,
+	// as well as how many ticks that corresponds to
+	// In the opposite direction (price < 1), we do the same thing (just decrement k instead of incrementing).
+	// The only difference is we must reduce the increment distance by a factor of 10.
+	if price.GT(sdk.OneDec()) {
+		for totalPrice.LT(price) {
+			curIncrement = handleNegativeExponents(currentK)
+			maxPriceForCurrentIncrement := kIncrementDistance.Mul(curIncrement)
+			totalPrice = totalPrice.Add(maxPriceForCurrentIncrement)
+			currentK = currentK.Add(sdk.OneInt())
+			ticksPassed = ticksPassed.Add(kIncrementDistance.TruncateInt())
+		}
+	} else {
+		for totalPrice.GT(price) {
+			curIncrement = handleNegativeExponents(currentK)
+			maxPriceForCurrentIncrement := kIncrementDistance.Mul(curIncrement)
+			totalPrice = totalPrice.Sub(maxPriceForCurrentIncrement)
+			currentK = currentK.Sub(sdk.OneInt())
+			ticksPassed = ticksPassed.Sub(kIncrementDistance.TruncateInt())
+		}
+	}
+	// Determine how many ticks we have passed in the current k increment
+	ticksToBeFulfilledByCurrentK := price.Sub(totalPrice).Quo(curIncrement)
+
+	// Finally, add the ticks we have passed from the completed k values, as well as the ticks we have passed in the current k value
+	tickIndex := ticksPassed.Add(ticksToBeFulfilledByCurrentK.TruncateInt())
+
+	// Add a check to make sure that the tick index is within the allowed range
+	minTick, maxTick := GetMinAndMaxTicksFromK(kAtPriceOne, kIncrementDistance)
+	if tickIndex.LT(minTick) {
+		return sdk.Int{}, fmt.Errorf("tickIndex must be greater than or equal to %s", minTick)
+	}
+	if tickIndex.GT(maxTick) {
+		return sdk.Int{}, fmt.Errorf("tickIndex must be less than or equal to %s", maxTick)
+	}
+
+	return tickIndex, nil
+}
+
+// handleNegativeExponents treats negative exponents as 1/(10**|exponent|) instead of 10**-exponent
+// This is because the sdk.Dec.Power function does not support negative exponents
+func handleNegativeExponents(exponent sdk.Int) sdk.Dec {
```

Comment:
It might be a more readable function name to call this `powTen` or something that indicates that it returns 10^(`exponent`)

Reply:
agreed, changed!